lib/AST/NameLookup.cpp

```diff
@@ -2812,28 +2872,53 @@ createOpaqueParameterGenericParams(
     if (!typeRepr)
       continue;
 
-    auto opaqueTypeReprs = typeRepr->collectOpaqueReturnTypeReprs();
-    for (auto opaqueRepr : opaqueTypeReprs) {
+    // Plain protocols should imply 'some' with experimetal feature
+    CollectedOpaqueReprs typeReprs;
+    if (ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
+        typeReprs = collectTypeReprs(typeRepr);
+    } else {  typeReprs = collectOpaqueReturnTypeReprs(typeRepr); }
+
+      for (auto repr : typeReprs) {
+   
+        if (isa<IdentTypeRepr>(repr)){
+            if(!isProtocol(evaluator,repr, ctx, dc))
+              continue;
+        }
       // Allocate a new generic parameter to represent this opaque type.
-      auto gp = GenericTypeParamDecl::create(
+        auto gp = GenericTypeParamDecl::create(
           dc, Identifier(), SourceLoc(), /*isTypeSequence=*/false,
           GenericTypeParamDecl::InvalidDepth, index++, /*isOpaqueType=*/true,
-          opaqueRepr);
-      gp->setImplicit();
+          repr);
+        gp->setImplicit();
 
       // Use the underlying constraint as the constraint on the generic parameter.
-      InheritedEntry inherited[1] = {
-        { TypeLoc(opaqueRepr->getConstraint()) }
-      };
-      gp->setInherited(ctx.AllocateCopy(inherited));
-
+      //  The underlying constraint is only present for OpaqueReturnTypeReprs
+        if( auto opaque = dyn_cast<OpaqueReturnTypeRepr>(repr)){
+              InheritedEntry inherited[1] = {
+                  { TypeLoc(opaque->getConstraint()) }
+              };
+              gp->setInherited(ctx.AllocateCopy(inherited));
+        } else {
+            InheritedEntry inherited[1] = {
+                { TypeLoc(repr) }
+            };
+            gp->setInherited(ctx.AllocateCopy(inherited));
+        }
       implicitGenericParams.push_back(gp);
     }
   }
 
   return implicitGenericParams;
 }
 
+
+bool swift::isProtocol( Evaluator &evaluator, TypeRepr *r, ASTContext &ctx, DeclContext *dc) {
+    DirectlyReferencedTypeDecls d = directReferencesForTypeRepr(evaluator, ctx, r, dc);
+    if(declsAreProtocols(d)){
```

Comment:
Since this is returning true if `declsAreProtocols(d)` is true and false otherwise, this can be simplified to just `return declsAreProtocols(d)`